cmd/ci_status.go

```diff
@@ -21,70 +40,177 @@ var ciStatusCmd = &cobra.Command{
+                }
+                fmt.Fprintln(w)
+        }
+
+        fmt.Fprintf(w, pipelineStatus(pipeline, jobs))
+        if wait && pipeline.Status != "success" {
+                os.Exit(1)
+        }
+        w.Flush()
+}
+
+func pipelineStatus(pipeline *gitlab.Pipeline, jobs []*gitlab.Job) string {
+        return fmt.Sprintf("\nPipeline Status:\t%s\n%s\n\n%s\n",
+                statusColor(pipeline.Status).Sprintf(pipeline.Status), timeMessage(pipeline), jobSummary(jobs))
+}
+
+func jobSummary(jobs []*gitlab.Job) string {
+        numPassed := 0
+        totalJobs := 0
+        numQueued := 0
+        numFailed := 0
+
+        for _, job := range jobs {
+                totalJobs++
+                if (job.Status == "success") {
+                  numPassed++
+                }
+                if (job.Status == "created") {
+                  numQueued++
+                }
+                if (job.Status == "failed") {
+                  numFailed++
+                }
+        }
+
+        return fmt.Sprintf("total\tpassed\tfailed\tqueued\n%d\t%d\t%d\t%d",
+                           totalJobs, numPassed, numFailed, numQueued)
+}
+
+func timeMessage(pipeline *gitlab.Pipeline) string {
+        if pipeline.Status == "pending" {
+          return fmt.Sprintf("created at %v", pipeline.CreatedAt)
+        } else if pipeline.Status == "running" {
+          return fmt.Sprintf("started at %v", pipeline.StartedAt)
+        } else {
+          hours   := pipeline.Duration / (60 * 60)
+          minutes := (pipeline.Duration / 60) - (hours * 60)
+          seconds := pipeline.Duration % 60
+          if hours > 0 {
+            return fmt.Sprintf("duration:\t%d hrs, %d min, %d secs", hours, minutes, seconds)
+          } else if minutes > 0 {
+            return fmt.Sprintf("duration:\t%d min, %d secs", minutes, seconds)
+          } else {
+            return fmt.Sprintf("duration:\t%d secs", pipeline.Duration)
+          }
+        }
+}
+
+func aliasFailures(f *flag.FlagSet, name string) flag.NormalizedName {
+	switch name {
+	case "failed":
+		name = "failures"
+		break
+	}
+	return flag.NormalizedName(name)
+}
+
+func statusColor(status string) *color.Color {
+        switch status {
+        case "failed":
+                return failed
+        case "success":
+                return passed
+        case "passed":
+                return passed
+        case "running":
+                return running
+        case "created":
+                return created
+        case "pending":
+                return created
+        case "skipped":
+                return skipped
+        default:
+                return defaultPrinter
+        }
 }
 
 func init() {
+        defaultPrinter.DisableColor()
 	ciStatusCmd.MarkZshCompPositionalArgumentCustom(1, "__lab_completion_remote_branches")
-	ciStatusCmd.Flags().Bool("wait", false, "Continuously print the status and wait to exit until the pipeline finishes. Exit code indicates pipeline status")
```

Comment:
It's preferred that we either use the above form here or create a config struct specific to this command to house the flags. This is because everything defined at the package level is shared between all of the other commands, and nami g collisions can cause subtle bugs